genesis/src/lib.rs

```diff
@@ -112,8 +112,12 @@ impl RawGenesisBlock {
             eprintln!("Genesis is quite large, it will take some time to apply it (size = {}, threshold = {})", size, Self::WARN_ON_GENESIS_GTE);
         }
         let reader = BufReader::new(file);
-        let mut raw_genesis_block: Self = serde_json::from_reader(reader)
-            .wrap_err_with(|| eyre!("Failed to deserialize raw genesis block from {:?}", &path))?;
+        let mut raw_genesis_block: Self = serde_json::from_reader(reader).wrap_err_with(|| {
+            eyre!(
+                "Failed to deserialize raw genesis block from {:?}",
+                path.as_ref().display()
```

Comment:
```suggestion
                "Failed to deserialize raw genesis block from {}",
                path.as_ref()
```

isn't this equivalent?

Reply:
It is somewhat different:

> A [`Path`](https://doc.rust-lang.org/std/path/struct.Path.html) might contain non-Unicode data. This struct [returned by `.display` implements the [`Display`](https://doc.rust-lang.org/std/fmt/trait.Display.html) trait in a way that mitigates that. It is created by the [`display`](https://doc.rust-lang.org/std/path/struct.Path.html#method.display) method on [`Path`](https://doc.rust-lang.org/std/path/struct.Path.html). This may perform lossy conversion, depending on the platform. If you would like an implementation which escapes the path please use [`Debug`](https://doc.rust-lang.org/std/fmt/derive.Debug.html) instead.

(from https://doc.rust-lang.org/std/path/struct.Display.html)